test/scripts/install-provisioner-working.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+# set -o pipefail
+set -o errexit
+set -o nounset
+
+readonly KIND=${KIND:-kind}
+readonly KUBECTL=${KUBECTL:-kubectl}
+
+readonly CLUSTERNAME=${CLUSTERNAME:-contour-e2e}
+readonly WAITTIME=${WAITTIME:-5m}
+
+readonly HERE=$(cd $(dirname $0) && pwd)
+readonly REPO=$(cd ${HERE}/../.. && pwd)
+
+kind::cluster::exists() {
+    ${KIND} get clusters | grep -q "$1"
+}
+
+kind::cluster::load::docker() {
+    ${KIND} load docker-image \
+        --name "${CLUSTERNAME}" \
+        "$@"
+}
+
+if ! kind::cluster::exists "$CLUSTERNAME" ; then
+    echo "cluster $CLUSTERNAME does not exist"
+    exit 2
+fi
+
+# Set the image tag to match the current Contour version.
+VERSION="v$$"
+
+# Build the Contour Provisioner image.
+make -C ${REPO} container IMAGE=ghcr.io/projectcontour/contour VERSION=${VERSION}
+
+# Push the Contour Provisioner image into the cluster.
+kind::cluster::load::docker ghcr.io/projectcontour/contour-provisioner:${VERSION}
+
+# Install the Contour Provisioner.
+${KUBECTL} apply -f ${REPO}/examples/gateway-provisioner/deploy.yaml
+${KUBECTL} apply -f ${REPO}/examples/gateway-provisioner/service.yaml
```

Comment:
The files you reference here do not exist.

Instead, the following should be apply'ed as-is:
- examples/gateway-provisioner/00-common.yaml
- examples/gateway-provisioner/01-roles.yaml
- examples/gateway-provisioner/02-rolebindings.yaml

For examples/gateway-provisioner/03-gateway-provisioner.yaml, you need to:
- change the imagePullPolicy to ifNotPresent
- set the tag of the image to the one we just built
- add an item to the args, `--contour-image=ghcr.io/projectcontour/contour:${VERSION}`
- apply the modified file

Please follow https://github.com/projectcontour/contour/blob/main/test/scripts/install-contour-working.sh#L72-L87 as an example.